### PR TITLE
added fancy annotationarray indexing

### PR DESCRIPTION
**Pull request by bmcfee**
Implements #112

- [x] documentation
- [x] unit tests

### CODE REVIEW

**Comment by bmcfee** [comment]
I think this is ready for CR.

@ejhumphrey ?

**Comment by bmcfee** [comment]
tumbleweed

File: jams/core.py [hunk 1]
```diff
@@ -1040,6 +1057,18 @@ def search(self, **kwargs):
 
         return results
 
+    def __getitem__(self, idx):
```
Comment (rabitt) [inline jams/core.py]: tricksy ;)

File: jams/core.py [hunk 2]
```diff
@@ -995,6 +995,23 @@ class AnnotationArray(list):
 
     This list subclass provides serialization and search/filtering
     for annotation collections.
+
+    Fancy-indexing can be used to directly search for annotations
+    belonging to a particular namespace.
+
+    Examples
```
Comment (rabitt) [inline jams/core.py]: These examples are helpful - is there a reasonable place where we could document more generically what kind of indexing is allowed?

**Comment by rabitt** [comment]
This lgtm. My only comment would be to add documentation somewhere that outlines exactly what kinds of indexing can be done, both generically and with the examples you already give.

**Comment by bmcfee** [comment]
Thanks for the CR.  I expanded the docs a little, should be good to go.

**Pull request merged**